python/agent.py

```diff
@@ -82,6 +82,58 @@ def browse_task_sandbox(agent_id: str, executor_sandbox_path: str, task_id: str)
         return []
 
 
+# Does download of files which match patterns in patterns_to_download. The files for download are located
+# in target dir specified in path_to_files.
+#
+# param: path_to_files  Relative path to target dir from task sandbox root. The path elements can be RE expressions,
+# marked by enclosing {}.
+# Example. For downloading log files in any location of kind '<sandbox-root>/nifi-<version>/logs/' you can specify
+#          this templated path: "{^nifi-(.+)$}/logs"
+def download_task_files_deep(agent_id: str, task_id: str, executor_sandbox_path: str, path_to_files: str,
+                             patterns_to_download: List[str], output_dir_base: str) -> int:
+    task_sandbox_path = os.path.join(executor_sandbox_path, "tasks/{}/".format(task_id))
+    task_sandbox = browse_agent_path(agent_id, task_sandbox_path)
+    if not task_sandbox:
+        logger.error("download_task_files_deep: cannot read task sandbox dir %s", task_sandbox_path)
+        return 3
+    output_dir_path = output_dir_base
+
+    for t in path_to_files.split('/'):
```

Comment:
It is a good practice to not use variable names like `t`, `x`, `m` etc. Let's give them some good names.

Reply:
Changed